### PR TITLE
Events x3

### PR DESCRIPTION
**Pull request by kelvin-cao**
- Added a variant variable, which is used together with gen variable for release mapping.
- Updated event related code for the added 'event supported' bit in event header in Gen4.

### CODE REVIEW

File: switchtec.c [hunk 1]
```diff
@@ -984,6 +1002,26 @@ static int event_ctl(struct switchtec_dev *stdev,
...
+static int event_supported(struct switchtec_dev *stdev, int eid, int idx)
+{
+	size_t off = event_regs[eid].offset;
+	u32 __iomem *hdr_reg;
+	u32 hdr;
+
+	if (stdev->gen == SWITCHTEC_GEN3) {
+		if (event_regs[eid].ver_mask &
+		    BIT(stdev->evlist_ver))
+			return 1;
```
Comment (lsgunth) [inline switchtec.c:1014]: I thought the consensus was to just leave GEN3 the way it was and not bother with masking off support for unsupported events. Gets rid of the need for firmware version specific stuff...

File: switchtec.c [hunk 2]
```diff
@@ -1568,6 +1568,19 @@ static enum switchtec_rel get_release(struct switchtec_dev *stdev)
 	return SWITCHTEC_REL_UNKNOWN;
 }
 
+static enum switchtec_evlist_ver get_evlist_ver(struct switchtec_dev *stdev)
+{
```
Comment (lsgunth) [inline switchtec.c:1616]: This seems rather unnecessary now.  The evlist is the same as the release.

File: switchtec.c (hunk 1 above)
Comment (kelvin-cao) [reply switchtec.c:1014]: I just want to distinguish the event list supported of psx/pfx and pax. I'll add it check in this function and remove the evlist related stuff.

File: switchtec.c (hunk 2 above)
Comment (kelvin-cao) [reply switchtec.c:1616]: Will remove. Do you think we need to check the release when probe? Like the driver currently support firmware releases through MR4, if customer use it on MR5, we quit in probe, or just assume the MR5 is supported?

File: switchtec.c (hunk 1 above)
Comment (kelvin-cao) [reply switchtec.c:1014]: I think it over, let's just keep all gen3 event stuff as is. More strict check may result in compatible issue of some PSX chip running PAX firmware, which is internally used.

Comment (lsgunth) [reply switchtec.c:1014]: Sounds good.

Comment (kelvin-cao) [reply switchtec.c:1014]: Given this, we don't even need var at this time. I'll remove it in the new PR.

**Pull request closed without merging**